src/middlewared/middlewared/plugins/network.py

```diff
@@ -155,6 +161,13 @@ def network_config_extend(self, data):
             Str('hostname_virtual', validators=[Match(r'^[a-zA-Z\.\-\0-9]+$')]),
             Str('domain', validators=[Match(r'^[a-zA-Z\.\-\0-9]+$')]),
             List('domains', items=[Str('domains')]),
+            Dict(
+                'service_announcement',
+                Bool('netbios', default=False),
+                Bool('mdns', default=True),
+                Bool('wsd', default=True),
+                register=True,
```

Comment:
Do we need register=True? I dont see that schema being referenced in other places, I may be msising something

Maybe what we want  here is update=True, so not all keys are populated if not provided?

Reply:
@william-gr will `update` work for inline schemas?

Reply:
Good question, I dont know, we should verify it

Reply:
It will not, it will be hard to implement, it will break PATCH restful method semantics and cause a lot of other confusion.

Reply:
Removed this.